lib/Optimizer/Transforms/QuakeSimplify.cpp

```diff
@@ -0,0 +1,313 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2025 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+#include "PassDetails.h"
+#include "cudaq/Optimizer/Dialect/Quake/QuakeOps.h"
+#include "cudaq/Optimizer/Transforms/Passes.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+#include "mlir/Transforms/Passes.h"
+
+namespace cudaq::opt {
+#define GEN_PASS_DEF_QUAKESIMPLIFY
+#include "cudaq/Optimizer/Transforms/Passes.h.inc"
+} // namespace cudaq::opt
+
+#define DEBUG_TYPE "quake-simplify"
+
+using namespace mlir;
+
+// Apply some simple quantum optimizations to quake. The quake operations are
+// expected to be in the value-semantics (having wire or control type operands).
+
+template <typename QOP>
+class HermitianElimination : public OpRewritePattern<QOP> {
+public:
+  using Base = OpRewritePattern<QOP>;
+  using Base::Base;
+
+  LogicalResult matchAndRewrite(QOP qop,
+                                PatternRewriter &rewriter) const override {
+    if (qop.getNegatedQubitControls())
```

Comment:
Don't we need to check for negatedQubitControls on the `prev` operation too?

Reply:
Yes. Good catch.

Reply:
Added f3b3046f